### DISCUSSION TITLE
Is there any ways to use graphql-ws in apollo-v2

### DISCUSSION THREAD

**Discussion by itezjaehyeon** [apollographql/apollo-server#6564] [Q&A]
I use `apollo-server-express` : `^2.25.3` 
but i hear that `subscriptions-transport-ws` will deprecate
so i want to switch it to graphql-ws but i dont know how to do this
please help...!!

**Comment by glasser** [comment]
https://www.apollographql.com/docs/apollo-server/data/subscriptions/ shows how to use `graphql-ws` with Apollo Server v3.

Using it with Apollo Server v2 isn't that different (you do have to disable AS2's own subscription support first) but we recommend you upgrade to AS3 first; the migration guide shows how to [keep subscriptions-transport-ws working with AS3](https://www.apollographql.com/docs/apollo-server/migration/#reenabling-subscriptions) and then you can switch your protocol later.
